kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala

```diff
@@ -148,6 +150,38 @@ trait ProcBuilder {
     proc
   }
 
+  def execKillYarnJob(appId: String): Int = {
+    val hadoopHomeOpt = env.get("HADOOP_HOME").orElse {
+      throw KyuubiSQLException(s"Failed to kill yarn job. HADOOP_HOME is not set! " +
+        "For more detail information on installing and configuring Yarn, please visit " +
+        "https://kyuubi.apache.org/docs/stable/deployment/on_yarn.html")
+    }.map { dir =>
+      Paths.get(dir, "bin", "yarn").toAbsolutePath.toFile.getCanonicalPath
+    }.get
+    val command = hadoopHomeOpt + " application -kill " + appId
+    val pb = new ProcessBuilder("/bin/sh", "-c", command)
+    var exitValue: Option[Int] = None
+
+    try {
+      val process = pb.start()
+      exitValue = Some(process.exitValue())
+    } catch {
+      case e: Exception =>
+        throw KyuubiSQLException(s"Failed to kill yarn job. Command[$command]", e.getCause)
+    }
+    info(s"Successfully kill yarn job $appId")
+    exitValue.get
+  }
+
+  private val APPLICATION_REGEX = Pattern.compile("application_\\d+_\\d+")
```

Comment:
`APPLICATION_REGEX` => `YARN_APP_NAME_REGEX`, prefer to use Scala regex, https://docs.scala-lang.org/tour/regular-expression-patterns.html